config/routes.rb

```diff
@@ -5,4 +5,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
- [OPTIONAL] Great job defining routes for your controller. Kindly check below snippet for more concise approach for defining route for this type of association. 

```
  
  resources :users, only: [:index, :show] do
      resources :posts, only: [:index, :show]
  end

```

Reply:
@Qoosim, thanks for the feedback, I'll look up more about this in order to improve the code.